RI.md

```diff
@@ -0,0 +1,842 @@
+
+### Output \Options
+
+#### Options `--format=FORMAT`, `-f FORMAT`
+
+Option `--format=FORMAT` (aliased as `-f`) specifies the formatter for the output,
+which must be `ansi`, `bs`, `markdown`, or `rdoc`;
+the default is `bs` for paged output, `ansi` otherwise.
+
+#### Options `--pager`, `--no-pager`
+
+Option `--pager` (the default) specifies that the output is to be piped
+to a pager;
+option `--no-pager` specifies that the output is not to be piped.
+
+#### Options `--width=NUMBER`, `-w NUMBER`
+
+Option `--width` (aliased as `-w`) specifies that the lengths of the displayed lines
+should be restricted to the given _NUMBER_ of characters;
+this is to be accomplished by line-wrapping, not truncation.
+The default width is `80`:
+
+```sh
+$ ri --width=40 Array | head
+= Array < Object
+
+----------------------------------------
+= Includes:
+Enumerable (from ruby core)
+
+(from ruby core)
+----------------------------------------
+An Array is an ordered, integer-indexed
+collection of objects, called
+```
+
+
+### List \Options
+
+#### Options `--list`, `-l`, `--no-list`
+
+Option `--list` (aliased as `-l`) specifies that all class and module names
+whose initial characters match the given _name_ are to be displayed:
+whose initial characters match the given _name_ are to be displayed:
+
+```sh
+$ ri --list Ar | head
+ArgumentError
+Array
+```
+
+If no _name_ is given, all class and module names are displayed.
+
+Option `--no-list` (the default) specifies that a list of class and module names
+is not to be displayed.
+
+### Methods \Options (for Class or Module)
+
+#### Options `--all`, `-a`, `--no-all`
+
+Option `--all` (aliased as `-a`) specifies that when _name_ identifies a class or module,
+the documents for all its methods are included;
+option `--no-all` (the default) specifies that the method documents are not to be included:
+
+```shell
+$ ri Array | wc -l
+390
+$ ri --all Array | wc -l
+4224
+```
+
+### Server Option
+
+#### Option `--server=NUMBER`
+
+Option `--server` specifies that the \RDoc server is to be run on the port
+given as _NUMBER_;
+the default port is `8214`.
+
+## Generating `ri` Source Files
+
+`ri` by default reads data from directories installed by Ruby and gems.
+
+You can create your own `ri` source files.
+This command creates `ri` source files in local directory `my_ri`,
+from Ruby source files in local directory `my_sources`:
+
+```sh
+$ rdoc --op my_ri --format=ri my_sources
+```
+
+Those files may then be considered for any `ri` command
+by specifying option `--doc-dir=my_ri`;
+see [option \\--doc-dir][20].
+
+[1]: rdoc-ref:RI.md@Static+Mode
+[2]: rdoc-ref:RI.md@Interactive+Mode
+[3]: rdoc-ref:RI.md@Class+and+Module+Documents
+[4]: rdoc-ref:RI.md@Options+--all-2C+-a-2C+--no-all
+[5]: rdoc-ref:RI.md@Method+Documents
+[6]: rdoc-ref:RI.md@Page+Documents
+[7]: rdoc-ref:RI.md@ri+Lists
+[8]: rdoc-ref:RI.md@Options+--list-2C+-l-2C+--no-list
+[9]: https://docs.ruby-lang.org/en/master/IRB.html
```

Comment:
IRB now has its own dedicated doc site, let's use it:

```suggestion
[9]: https://ruby.github.io/irb/
```

Reply:
I disagree.  I'm writing for Ruby, and prefer to stay on its site when practicable.